cmd/kubeadm/app/images/images.go

```diff
@@ -32,7 +33,8 @@ func GetGenericImage(prefix, image, tag string) string {
 // GetKubernetesImage generates and returns the image for the components managed in the Kubernetes main repository,
 // including the control-plane components and kube-proxy. If specified, the HyperKube image will be used.
 func GetKubernetesImage(image string, cfg *kubeadmapi.ClusterConfiguration) string {
-	if cfg.UseHyperKubeImage {
+	if cfg.UseHyperKubeImage && image != constants.HyperKube {
+		klog.Warningf(`WARNING: DEPRECATED use of the "hyperkube" image in place of %q`, image)
```

Comment:
When this warning will be printed? I see a bit of convoluted logic here at `images.go` because we'll never satisfy both conditions, as `GetControlPlaneImages` has a explicit branch for `cfg.UseHyperKubeImage` as well. Is this fine?

Reply:
Ah, this looks for other usages like in `controlplane/manifests.go` and `addons/proxy/proxy.go`, right?

Reply:
Yep, actually the messages will be dumped several times for the different images and probably as a whole.

Reply:
That is fine IMO.